pkg/sdk/testint/tasks_gen_integration_test.go

```diff
@@ -592,4 +592,135 @@ func TestInt_Tasks(t *testing.T) {
+		t.Cleanup(func() {
+			require.NoError(t, client.Tasks.Alter(ctx, sdk.NewAlterTaskRequest(rootTask.ID()).WithSuspend(sdk.Bool(true))))
+		})
+
+		tasksToResume, err := client.Tasks.SuspendRootTasks(ctx, task.ID(), task.ID())
+		require.NoError(t, err)
+		require.NotEmpty(t, tasksToResume)
+		require.Contains(t, tasksToResume, rootTask.ID())
+
+		rootTaskStatus, err := client.Tasks.ShowByID(ctx, rootTask.ID())
+		require.NoError(t, err)
+		require.Equal(t, sdk.TaskStateSuspended, rootTaskStatus.State)
+
+		middleTaskStatus, err := client.Tasks.ShowByID(ctx, middleTask.ID())
+		require.NoError(t, err)
+		require.Equal(t, sdk.TaskStateStarted, middleTaskStatus.State)
+
+		require.NoError(t, client.Tasks.ResumeTasks(ctx, tasksToResume))
+
+		rootTaskStatus, err = client.Tasks.ShowByID(ctx, rootTask.ID())
+		require.NoError(t, err)
+		require.Equal(t, sdk.TaskStateStarted, rootTaskStatus.State)
+
+		middleTaskStatus, err = client.Tasks.ShowByID(ctx, middleTask.ID())
+		require.NoError(t, err)
+		require.Equal(t, sdk.TaskStateStarted, middleTaskStatus.State)
+	})
+
+	t.Run("suspend root tasks temporarily with three sequentially connected tasks - middle in DAG", func(t *testing.T) {
+		rootTaskId := sdk.NewSchemaObjectIdentifier(testDb(t).Name, testSchema(t).Name, random.String())
+		rootTask := createTaskWithRequest(t, sdk.NewCreateTaskRequest(rootTaskId, sql).WithSchedule(sdk.String("60 minutes")))
+
+		middleTaskId := sdk.NewSchemaObjectIdentifier(testDb(t).Name, testSchema(t).Name, random.String())
+		middleTask := createTaskWithRequest(t, sdk.NewCreateTaskRequest(middleTaskId, sql).WithAfter([]sdk.SchemaObjectIdentifier{rootTask.ID()}))
+
+		childTaskId := sdk.NewSchemaObjectIdentifier(testDb(t).Name, testSchema(t).Name, random.String())
+		childTask := createTaskWithRequest(t, sdk.NewCreateTaskRequest(childTaskId, sql).WithAfter([]sdk.SchemaObjectIdentifier{middleTask.ID()}))
+
+		require.NoError(t, client.Tasks.Alter(ctx, sdk.NewAlterTaskRequest(childTask.ID()).WithResume(sdk.Bool(true))))
+		t.Cleanup(func() {
+			require.NoError(t, client.Tasks.Alter(ctx, sdk.NewAlterTaskRequest(childTask.ID()).WithSuspend(sdk.Bool(true))))
+		})
+
+		require.NoError(t, client.Tasks.Alter(ctx, sdk.NewAlterTaskRequest(rootTask.ID()).WithResume(sdk.Bool(true))))
+		t.Cleanup(func() {
+			require.NoError(t, client.Tasks.Alter(ctx, sdk.NewAlterTaskRequest(rootTask.ID()).WithSuspend(sdk.Bool(true))))
+		})
+
+		tasksToResume, err := client.Tasks.SuspendRootTasks(ctx, middleTask.ID(), middleTask.ID())
+		require.NoError(t, err)
+		require.NotEmpty(t, tasksToResume)
+
+		rootTaskStatus, err := client.Tasks.ShowByID(ctx, rootTask.ID())
+		require.NoError(t, err)
+		require.Equal(t, sdk.TaskStateSuspended, rootTaskStatus.State)
+
+		childTaskStatus, err := client.Tasks.ShowByID(ctx, childTask.ID())
+		require.NoError(t, err)
+		require.Equal(t, sdk.TaskStateStarted, childTaskStatus.State)
```

Comment:
hmmm, this is interesting that the child task is not suspended automatically when the root task is. But I suppose this is how SF works.

Reply:
Yup yup